examples/3dsphere/driver.jl

```diff
@@ -0,0 +1,97 @@
+using Logging: global_logger
+using TerminalLoggers: TerminalLogger
+global_logger(TerminalLogger())
+
+import ClimaCore: enable_threading
+enable_threading() = true
+
+using OrdinaryDiffEq
+
+const FT = Float64
+
+default_test_name = "baroclinic_wave_ρe"
+test_implicit_solver = false # makes solver extremely slow when set to `true`
+
+################################################################################
+
+if haskey(ENV, "TEST_NAME")
+    test_name = ENV["TEST_NAME"]
+else
+    test_name = default_test_name
+end
+
+include("utilities.jl")
+include("$test_name.jl")
+
+@unpack zmax,
+velem,
+helem,
+npoly,
+tmax,
+dt,
+ode_algorithm,
+jacobian_flags, # relevant for implicit/IMEX ODE algorithms
+max_newton_iters, # relevant for ODE algorithms that use Newton's method
+save_every_n_steps,
+additional_solver_kwargs = driver_values(FT)
+
+center_local_geometry, face_local_geometry =
+    local_geometry_fields(FT, zmax, velem, helem, npoly)
+Yc = map(initial_condition, center_local_geometry)
+uₕ = map(initial_condition_velocity, center_local_geometry)
+w = map(_ -> Geometry.Covariant3Vector(FT(0.0)), face_local_geometry)
+Y = Fields.FieldVector(Yc = Yc, uₕ = uₕ, w = w)
+p = merge(implicit_cache_values(Y, dt), remaining_cache_values(Y, dt))
+
+use_transform = !(ode_algorithm in (Rosenbrock23, Rosenbrock32))
+W_kwarg = use_transform ? (; Wfact_t = Wfact!) : (; Wfact = Wfact!)
+W = SchurComplementW(Y, use_transform, jacobian_flags, test_implicit_solver)
+problem = SplitODEProblem(
+    ODEFunction(
+        implicit_tendency!;
+        W_kwarg...,
+        jac_prototype = W,
+        tgrad = (dY, Y, p, t) -> fill!(dY, FT(0.0)),
```

Comment:
```suggestion
        tgrad = (dY, Y, p, t) -> fill!(dY, FT(0.0)),
```
Let's avoid `fill!` here since it seems to be allocating a lot (it may still be falling back to `getindex` for FieldVectors).